### PR TITLE
Fast fix for the drop up menu bug #4341 

### PR DESCRIPTION
**Pull request by flegmaatikko**
Here is my first solution for the bug #4341 
I created a new method that shows where the target button is and did minor changed to the texts. There is also an unused method that opens the menu only. It can be removed if needed.

The mouse pointer moves over the menu but not over to the button. Instead of moving the mouse pointer I tried to show the position of target menu item by highlighting the buttons step by step from bottom to up.

This is just a fist draft and I look forward to get some feedback.

### CODE REVIEW

**Comment by gunchleoc** [comment]
I like it, you can continue like that for the other dropdowns in this turorial :)

**Comment by flegmaatikko** [comment]
Hopefully all the cases are fixed now.

File: data/campaigns/tutorial01_basic_control.wmf/scripting/helper_functions_demonstration.lua [hunk 1]
```diff
@@ -71,12 +71,25 @@ end
...
+   blocker:lift_blocks()
+end
 
+function open_item_from_dropdown(name, item)
+   wl.ui.MapView().dropdowns[name]:open(item)
+end
+
+function show_item_from_dropdown(name, item)
+   local blocker = UserInputDisabler:new()
+   local ind = wl.ui.MapView().dropdowns[name]:get_no_of_items()
```
Comment (gunchleoc) [inline data/campaigns/tutorial01_basic_control.wmf/scripting/helper_functions_demonstration.lua]:
```suggestion
   local ind = wl.ui.MapView().dropdowns[name].no_of_items
```

File: src/scripting/lua_ui.cc [hunk 2]
```diff
@@ -367,6 +367,7 @@ const MethodType<LuaDropdown> LuaDropdown::Methods[] = {
    METHOD(LuaDropdown, open),
    METHOD(LuaDropdown, highlight_item),
    METHOD(LuaDropdown, select),
+   METHOD(LuaDropdown, get_no_of_items),
```
Comment (gunchleoc) [inline src/scripting/lua_ui.cc]: Remove this. It's available as a property and we don't need to add a method too.

**Review by gunchleoc — CHANGES_REQUESTED** [review]

**Review by gunchleoc — APPROVED** [review]

**Pull request merged**